db/src/main/java/io/lantern/db/DB.kt

```diff
@@ -745,12 +791,21 @@ open class Queryable internal constructor(
         count: Int,
         fullTextSearch: String?,
         reverseSort: Boolean,
+        fullTextHighlightStartDelim: String = "*",
+        fullTextHighlightEndDelim: String = "*",
         onRow: (cursor: Cursor) -> Unit
     ) {
         val cursor = if (fullTextSearch != null) {
             db.rawQuery(
-                "SELECT d.path, d.value FROM ${schema}_fts f INNER JOIN ${schema}_data d ON f.rowid = d.rowid WHERE d.path LIKE ? AND f.value MATCH ? ORDER BY f.rank LIMIT ? OFFSET ?",
-                arrayOf(serde.serialize(pathQuery), fullTextSearch, count, start)
+                "SELECT d.path, d.value, highlight(${schema}_fts2, 0, ?, ?) FROM ${schema}_fts2 f INNER JOIN ${schema}_data d ON f.rowid = d.rowid WHERE d.path LIKE ? AND f.value MATCH ? ORDER BY f.rank LIMIT ? OFFSET ?",
+                arrayOf(
```

Comment:
@oxtoacart Do you mind explaining the `ORDER BY f.rank`? (I was trying to find with what order logic the results come back to the client)

Reply:
Yes, f.rank is a virtual column that indicates how well the row matches the query. So what this means is that results are returned in the order of best to least best match.